### PR TITLE
Remove SCS specific scaling for PSD constraints

### PR DESCRIPTION
**Pull request by AKS1996**
As learned in https://github.com/jump-dev/MatrixOptInterface.jl/pull/7, code for MatOI, MOSD, and DiffOpt contained a lot of SCS specific code - which is now removed 

Since I was comparing DiffOpt results by diffcp, which was SCS styled. But I hand-derived a few examples (duals, slack variables and their derivatives) and now the updated gradients are much simpler to understand

### CODE REVIEW

**Comment by AKS1996** [comment]
I'm copying whole projections.jl in DiffOpt. Will remove it once this PR is mergeg

File: src/projections.jl [hunk 1]
```diff
@@ -76,46 +76,53 @@ end
...
-            X[j, i] *= √2
-        end
-    end
-    return X[LinearAlgebra.tril(trues(size(X)))]
+#     @inbounds for i in 1:size(X)[1]
+#         for j in i+1:size(X)[2]
+#             X[i, j] *= √2
+#             X[j, i] *= √2
+#         end
+#     end
```
Comment (matbesancon) [inline src/projections.jl:124]:
```suggestion
```

File: src/projections.jl [hunk 2]
```diff
@@ -211,7 +212,7 @@ function projection_gradient_on_set(::DefaultDistance, v::AbstractVector{T}, ::M
     λ, U = LinearAlgebra.eigen(X)
 
     # if all the eigenvalues are >= 0
-    if all(>=(0), λ)
+    if all(λ .≥ zero(T))
```
Comment (matbesancon) [inline src/projections.jl:215]: why this change? This seems to be allocating when the first version was not

Comment (AKS1996) [reply src/projections.jl:215]: strangely this breaks on my desktop now (worked earlier) and in this test too - https://travis-ci.com/github/matbesancon/MathOptSetDistances.jl/jobs/427137353

Comment (matbesancon) [inline src/projections.jl:215]:
```suggestion
    if all(λi ≥ zero(λi) for λi in λ)
```

Comment (matbesancon) [reply src/projections.jl:215]: a generator will boil down to a for loop and avoid the allocation of a broadcast

**Comment by matbesancon** [comment]
Looks good! will merge when Travis passes

**Review by matbesancon — APPROVED** [review]

**Comment by matbesancon** [comment]
Thanks!
